scripts/element_tracking.py

```diff
@@ -0,0 +1,68 @@
+import contextlib
+import json
+from pathlib import Path
+from types import TracebackType
+from typing import (
+    Callable,
+    Dict,
+    Iterator,
+    MutableMapping,
+    Optional,
+    Set,
+    Type,
+)
+
+
+class ElementsCache:
+    def __init__(self, cache_path: Path) -> None:
+        self.cache_path = cache_path
+        self.elements: Dict[str, int] = {}
+
+        if cache_path.exists():
```

Comment:
This has a race condition between the test and the read—might be better to catch the exception.

Reply:
aaf3b7d